File: leetcode/0062-unique-paths/solution.py
class Solution(object):
    def uniquePaths(self, m, n):
        """
        :type m: int
        :type n: int
        :rtype: int
        """
        # Create a matrix of all zeroes to represent the unique paths
        # We have padding so we don't have to deal with border detection
        matrix = [ [0 for i in range(n+1)] for j in range(m+1) ]
        
        
        # This is a hack to get a 1 in the square at 1,1
        matrix[0][1] = 1
        # print(matrix)
        
        for row in range(1, m+1):
            for col in range(1, n+1):
                # We progress first left to right, then top to bottom across the matrix
                # From the cells directly above and to the left, there is one way to get to this cell
                #       therefore we simply add them together
                matrix[row][col] = matrix[row-1][col] + matrix[row][col-1]
        
        # print(matrix)
        return matrix[-1][-1]
                
        

File: leetcode/0062-unique-paths/solution.go
func uniquePaths(m int, n int) int {
	// Create an m by n matrix of 0s
	dp := make([][]int, m)
	for i := 0; i < m; i++ {
		dp[i] = make([]int, n)
	}

	// The top left corner is set to 1
	dp[0][0] = 1

	// Set the top-most row and left-most column to 1's, since there's always
	// only one way to get there.
	for i := range dp {
		dp[i][0] = 1
	}

	for i := range dp[0] {
		dp[0][i] = 1
    }

	for i := 1; i < m; i++ {
        for j := 1; j < n; j++ {
            dp[i][j] = dp[i-1][j] + dp[i][j-1]
        }
    }

	return dp[m-1][n-1]

}
